admin/server/api/list/download.js

```diff
@@ -33,32 +33,16 @@ module.exports = function (req, res) {
 	}
 	var sort = req.list.expandSort(req.query.sort);
 	query.sort(sort.string);
-	query.exec(function (err, results) {
+	query.exec()
+	.then(function (results) {
 		var data;
-		var fields = [];
-		if (err) return res.apiError('database error', err);
 		if (format === 'csv') {
 			data = results.map(function (item) {
-				var row = req.list.getCSVData(item, {
-					expandRelationshipFields: req.query.expandRelationshipFields,
-					fields: req.query.select,
-					user: req.user,
-				});
-				// If nested values in the first item aren't present, babyparse
-				// won't add them even if they are present in others. So we
-				// add keys from all items to an array and explicitly provided
-				// the complete set to baby.unparse() below
-				Object.keys(row).forEach(function (i) {
-					if (fields.indexOf(i) === -1) fields.push(i);
-				});
-				return row;
+				return req.list.getCSV(item, req.query.select, req.query.expandRelationshipFields);
```

Comment:
it also doesn't look like the current function signature for https://github.com/keystonejs/keystone/blob/c4789388ef09d09f39585f68b2a6fc898b7da040/lib/list/getCSVData.js#L30